doc/environ.sh.sample

```diff
@@ -1,102 +1,173 @@
-# using arm-eabi as the target, rather than arm-elf. dc-chain now defaults to
-# arm-eabi, so that's the default here.
-#export DC_ARM_BASE="/usr/local/dc/arm-elf"
-#export DC_ARM_PREFIX="arm-elf"
-export DC_ARM_BASE="/opt/toolchains/dc/arm-eabi"
-export DC_ARM_PREFIX="arm-eabi"
+# Make Utility
+#
+# Configures the tool to be used as the main "make" utility
+# for building GNU Makefiles. On a platform such as BSD, 
+# the default can be changed to "gmake," for the GNU 
+# implementation.
+#
+export KOS_MAKE="make"
+#export KOS_MAKE="gmake"
 
-# Expand PATH if not already set (comment out if you don't want this done here)
-if [[ ":$PATH:" != *":${KOS_CC_BASE}/bin:/opt/toolchains/dc/bin"* ]]; then
-  export PATH="${PATH}:${KOS_CC_BASE}/bin:/opt/toolchains/dc/bin"
-fi
+# Loader Utility
+#
+# Specifies the loader to be used with the "make run" targets
+# in the KOS examples. Defaults to using a preconfigured version
+# of dc-tool. Use one of the other options for a manual dc-tool-ip
+# or dc-tool-serial configuration, remembering to change the values
+# for the Dreamcast's IP address or the serial port interface.
+#
+export KOS_LOADER="dc-tool -x"				       
+#export KOS_LOADER="dc-tool-ip -t 192.168.1.100 -x"
+#export KOS_LOADER="dc-tool-ser -t /dev/ttyS0 -x" 
 
-# reset some options because there's no reason for them to persist across
-# multiple sourcing of this
+# Default Compiler Flags
+#
+# Resets build flags. You can also initialize them to some preset
+# default values here if you wish.
+#
 export KOS_INC_PATHS=""
 export KOS_CFLAGS=""
 export KOS_CPPFLAGS=""
 export KOS_LDFLAGS=""
 export KOS_AFLAGS=""
 export DC_ARM_LDFLAGS=""
 
-# Setup some default CFLAGS for compilation. The things that will go here
-# are user specifyable, like optimization level and whether you want stack
-# traces enabled. Some platforms may have optimization restrictions,
-# please check README.
-# GCC seems to have made -fomit-frame-pointer the default on many targets, so
-# hence you may need -fno-omit-frame-pointer to actually have GCC spit out frame
-# pointers. It won't hurt to have it in there either way.
-# Link-time optimizations can be enabled by adding -flto. It however requires a
-# recent toolchain (GCC 10+), and has not been thoroughly tested.
-export KOS_CFLAGS="-O2 -fomit-frame-pointer"
-# export KOS_CFLAGS="-O2 -DFRAME_POINTERS -fno-omit-frame-pointer"
+# Optimization Level 
+#
+# Controls the baseline optimization level to use when building. 
+# Typically this is -Og (debugging), -O0, -01, -02, or -03.
+# NOTE: For our target, -O4 is a valid optimization level that has 
+# been seen to have some performance impact as well.
+#
+export KOS_CFLAGS="${KOS_CFLAGS} -O2"
+
+# Additional Optimizations
+#
+# Uncomment this to enable what has been found emperically to be 
+# the optimal set of additional flags for release build performance 
+# on the current stable toolchain.
+#
+#export KOS_CFLAGS="${KOS_CFLAGS} -freorder-blocks-algorithm=simple -flto=auto"
```

Comment:
Since certain kos-ports and examples won't work with `-flto=auto` that should be mentioned here.

Reply:
Good point!